### PR TITLE
Validate user key for whitespace chars

### PR DESCRIPTION
**Pull request by koln67**
Add validation and testing for whitespace in user string. If whitespace is detected in the user string, it should be considered an invalid user string because it could be exploiting a security loophole.

### CODE REVIEW

File: utils/main.go [hunk 1]
```diff
@@ -88,6 +89,20 @@ func CheckExpired(expireOn string) (bool, error) {
 
 }
 
+// ValidateUserKey checks for the presence of a characters which should not be
+// allowed in a username string, returns an error if any such characters are
+// detected, nil otherwise.
+// Currently, the only banned characters are whitespace characters.
+func ValidateUserKey(user string) (error) {
+	whiteSpaceRegexp, _ := regexp.Compile("\\s")
```
Comment (dorileo) [inline utils/main.go:97]: Do you need to use `regexp` for this purpose? Wont `strings.Contains()` be more efficient?

File: utils/main_test.go [hunk 2]
```diff
@@ -83,3 +83,25 @@ func TestCheckExpiredKey(t *testing.T) {
...
+		{"user -g 27", false},
+		{"user\t-g", false},
+		{"user\n-g", false},
+		{"username\t-g\n27", false},
+	}
+	for _, tt := range table {
+		err := ValidateUserKey(tt.user)
+		isValid := err != nil
+		if isValid != tt.valid {
+			t.Errorf("ValidateUserKey(%s) incorrect return: valid: %t - want valid: %t", tt.user, isValid, tt.valid)
```
Comment (dorileo) [inline utils/main_test.go:104]: I'd rephrase it as: `"Invalid ValidateUserKey(%s) return: expected: %t - got: %t"`

**Review by dorileo — COMMENTED** [review]
Minor comments...

File: utils/main.go (hunk 1 above)
Comment (koln67) [reply utils/main.go:97]: Does strings.Contains() have a way to check for any whitespace character? 

Strictly speaking, it seems like anything which falls under unicode.IsSpace() should not be in the user string.

Comment (dorileo) [reply utils/main.go:97]: I see, looks good then.

File: utils/main.go [hunk 3]
```diff
@@ -88,6 +89,20 @@ func CheckExpired(expireOn string) (bool, error) {
...
 
+// ValidateUserKey checks for the presence of a characters which should not be
+// allowed in a username string, returns an error if any such characters are
+// detected, nil otherwise.
+// Currently, the only banned characters are whitespace characters.
+func ValidateUserKey(user string) error {
+	whiteSpaceRegexp, _ := regexp.Compile("\\s")
+
+	hasWhiteSpace := whiteSpaceRegexp.MatchString(user)
+	if hasWhiteSpace {
```
Comment (hopkiw) [inline utils/main.go:100]: these lines can be combined

File: utils/main_test.go (hunk 2 above)
Comment (koln67) [reply utils/main_test.go]: Changed

File: utils/main.go (hunk 3 above)
Comment (koln67) [reply utils/main.go]: Changed: now reads with hasWhiteSpace == true for readability

File: utils/main.go [hunk 4]
```diff
@@ -88,6 +89,19 @@ func CheckExpired(expireOn string) (bool, error) {
...
 }
 
+// ValidateUserKey checks for the presence of a characters which should not be
+// allowed in a username string, returns an error if any such characters are
+// detected, nil otherwise.
+// Currently, the only banned characters are whitespace characters.
+func ValidateUserKey(user string) error {
+	whiteSpaceRegexp, _ := regexp.Compile("\\s")
+
+	if hasWhiteSpace := whiteSpaceRegexp.MatchString(user); hasWhiteSpace == true {
```
Comment (dorileo) [inline utils/main.go:99]: Are you using `hasWhiteSpace` elsewhere?

Why not simply doing the following(I guess this is what @hopkiw meant):

```go
if whiteSpaceRegexp.MatchString(user) {
```

Comment (koln67) [reply utils/main.go]: fixed

**Review by hopkiw — APPROVED** [review]
